### PR TITLE
add value to the model if not exist

### PR DESCRIPTION
**Pull request by hutnikau**
related to:
https://github.com/oat-sa/qti-sdk/pull/22

### CODE REVIEW

File: views/js/core/databinder.js [hunk 1]
```diff
@@ -253,7 +253,7 @@ function($, _, Handlebars, Encoders, Filters){
      */
     DataBinder.prototype._bindNode = function _bindNode($node, path, model, domFirst) {
         if(!$node.data('bound')){
-            if(domFirst === true){
+            if(domFirst === true || locate(model, path) === undefined){
```
Comment (ssipasseuth) [inline views/js/core/databinder.js]: please add a new unit test to illustrate this bug fix

Comment (hutnikau) [inline views/js/core/databinder.js]: I created unit test. Please review it again.

File: views/js/test/core/databinder/test.js [hunk 2]
```diff
@@ -182,6 +182,26 @@ define(['jquery', 'core/databinder'], function($, DataBinder){
...
+        var $sectionParts = $('ul', $container);
+        assert.ok($container.length === 1, 'Test the fixture is available');
+
+        var $newSection = $("<li>" +
+            "<input checked='checked' type='checkbox' value='true' data-bind-encoder='boolean' data-bind='visible'>" +
+            "</li>");
+
+        $sectionParts.append($newSection);
+
+        databinder = new DataBinder($container, model);
```
Comment (ssipasseuth) [inline views/js/test/core/databinder/test.js]: that is a global var ...

Comment (hutnikau) [inline views/js/test/core/databinder/test.js]: Oops! Fixed.

File: views/js/test/core/databinder/test.js [hunk 3]
```diff
@@ -182,6 +182,26 @@ define(['jquery', 'core/databinder'], function($, DataBinder){
...
+        var $newSection = $("<li>" +
+            "<input checked='checked' type='checkbox' value='true' data-bind-encoder='boolean' data-bind='visible'>" +
+            "</li>");
+
+        $sectionParts.append($newSection);
+
+        var databinder = new DataBinder($container, model);
+        databinder.bind();
+
+        strictEqual(model.testParts[0].assessmentSections[0].sectionParts.length, 4, 'model length has been updated');
```
Comment (ssipasseuth) [inline views/js/test/core/databinder/test.js]: ok, now I am still trying to understand.... but it seems that the model lenght has not been update : the length was initially 4, wasn't it ?
If you remove the change you made on  databinder.js, this condition still evaluate to true, is that expected ?

Comment (ssipasseuth) [inline views/js/test/core/databinder/test.js]: also, I am wondering, should you not bind using domFirst=true first to solve this issue ?

Comment (hutnikau) [inline views/js/test/core/databinder/test.js]: You are right. This assertion is redundant.
I removed it.

Comment (hutnikau) [inline views/js/test/core/databinder/test.js]: `domFirst=true` will not solve this issue. At the moment when property panel will be added to the DOM model (for example `identifier` property) of this panel is already exists, but all inputs in added panel is empty. So, if we set `domFirst=true` then all existing properties in the model will be erased. You can try it by setting `databinder.bind(null, null, null, true);` here:
https://github.com/oat-sa/extension-tao-testqti/blob/develop/views/js/controller/creator/views/property.js#L58

**Pull request merged**